### ISSUE TITLE
SC1094 should be an error, not just a warning

### ISSUE THREAD

**Issue by Whissi** [koalaman/shellcheck#1916]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1094
- My shellcheck version (`shellcheck --version` or "online"): 0.7.1
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC1094)
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:
main.sh:
```sh
#!/bin/sh
. lib.sh
```

lib.sh:
```sh
#!/bin/sh

if foo
blah
```

Command: `shellcheck --check-sourced --severity=error --external-sources main.sh`

#### Here's what shellcheck currently says:
No output. Exit code 0.

#### Here's what I wanted or expected to see:
```
  ^-----------------^ SC1094: Parsing of sourced file failed.
```
Exit code 1.

### Rational
At the moment I am using shellcheck to guard against errors only. I.e. I want to prevent any bad commit which could cause the script itself to fail because of a syntax error or something like that.

I am really surprised seeing that SC1094 is just warning in this scenario.

When I only test lib.sh, I'll get a fatal error like expected.

Maybe I am missing something but if you don't agree, an option allowing me to change severity for specific rules would also work for me.
